File: plugins/agent-utilities/skills/fusion-parametric-design/src/fusion_design/enclosure_features/evidence.py
"""Evidence labels for enclosure-feature dimensions; no material database."""

from enum import Enum


class EvidenceClass(str, Enum):
    GEOMETRIC_INVARIANT = "geometric-invariant"
    FUSION_API_CONSTRAINT = "fusion-api-constraint"
    MANUFACTURER_SPECIFIED = "manufacturer-specified"
    STANDARD_SPECIFIED = "standard-specified"
    MATERIAL_DATASHEET = "material-datasheet"
    FDM_PROCESS_HEURISTIC = "fdm-process-heuristic"
    USER_PREFERENCE = "user-preference"
    PROVISIONAL_DEFAULT = "provisional-default"
    COUPON_VERIFIED = "coupon-verified"
    PHYSICAL_TEST_REQUIRED = "physical-test-required"


SOURCE_REQUIRED = {
    EvidenceClass.MANUFACTURER_SPECIFIED,
    EvidenceClass.STANDARD_SPECIFIED,
    EvidenceClass.MATERIAL_DATASHEET,
    EvidenceClass.COUPON_VERIFIED,
}


def validate_evidence(evidence_class: EvidenceClass, source_ref: str | None) -> None:
    if not isinstance(evidence_class, EvidenceClass):
        raise ValueError("evidence_class must be a supported EvidenceClass")
    if evidence_class in SOURCE_REQUIRED and not (source_ref and source_ref.strip()):
        raise ValueError(f"{evidence_class.value} requires source_ref")

File: plugins/agent-utilities/skills/fusion-parametric-design/src/enclosure-features/evidence.ts
/** Closed evidence taxonomy for enclosure feature rules. */

export const EVIDENCE_CLASSES: ReadonlySet<string> = new Set([
  "geometric-invariant",
  "fusion-api-constraint",
  "manufacturer-specified",
  "standard-specified",
  "material-datasheet",
  "fdm-process-heuristic",
  "user-preference",
  "provisional-default",
  "coupon-verified",
  "physical-test-required",
]);

/**
 * The exact fabrication changes that stale a process-sensitive rule. A rule's
 * four boolean flags select from this set; the names are part of the contract.
 */
export const INVALIDATION_FLAGS = [
  "invalidated_by_material_change",
  "invalidated_by_nozzle_change",
  "invalidated_by_layer_height_change",
  "invalidated_by_orientation_change",
] as const;
